server/src/main/java/io/druid/server/http/InventoryViewUtils.java

```diff
@@ -37,42 +33,22 @@
 import java.util.Set;
 import java.util.TreeSet;
 
-public class InventoryViewUtils
+public interface InventoryViewUtils
 {
 
-  public static Set<DruidDataSource> getDataSources(InventoryView serverInventoryView)
+  static Set<ImmutableDruidDataSource> getDataSources(InventoryView serverInventoryView)
   {
-    TreeSet<DruidDataSource> dataSources = Sets.newTreeSet(
-        new Comparator<DruidDataSource>()
-        {
-          @Override
-          public int compare(DruidDataSource druidDataSource, DruidDataSource druidDataSource1)
-          {
-            return druidDataSource.getName().compareTo(druidDataSource1.getName());
-          }
-        }
-    );
-    dataSources.addAll(
-        Lists.newArrayList(
-            Iterables.concat(
-                Iterables.transform(
-                    serverInventoryView.getInventory(),
-                    new Function<DruidServer, Iterable<DruidDataSource>>()
-                    {
-                      @Override
-                      public Iterable<DruidDataSource> apply(DruidServer input)
-                      {
-                        return input.getDataSources();
-                      }
-                    }
-                )
-            )
-        )
-    );
-    return dataSources;
+    return serverInventoryView.getInventory().stream()
+                              .flatMap(server -> server.getDataSources().stream())
```

Comment:
Same comment as before about suggesting a `.map` followed by a `flatMap`

Reply:
```flatMap``` should be followed by ```map``` here because ```server.getDataSources()``` returns a ```Collection<DruidDataSource>``` which should be flatten before applying ```map```.